src/infer_single_paired.py

```diff
@@ -0,0 +1,382 @@
+import os
+import sys
+import gzip
+import subprocess as sp
+from itertools import islice
+sys.tracebacklimit = 0
+
+messages = {
+    1: "Single End or First_mate \n",
+    2: "Second_mate \n",
+    3: "Mixed \n",
+    5: "First_mate or Second_mate \n"
+}
+
+
+class End_parser:
+
+    """ Infer Single or Paired """
+
+    def __init__(self):
+        print("\nRunning Parser\n")
+
+    def SRA_contents(self, name, filename):
+
+        """ Runs fastq-dump """
```

Comment:
In this function I am not using fastq-dump to split into two Fastq files, as I know if I will do that, resulting files will be very large. All I am doing is, after taking input file from the user, and if it's identifier matches to the SRA format, then I am using fastq-dump to generate just at max 8 lines ( 1 whole spot , that too at as a subprocess , so there is no new resulting file), what this will do is :
1. This will first do a normal check , even though if somehow file's identifier matches the SRA format, but the experiment type is not there in the SRA archive , it will throw an error.
2. Since after returning , one whole spot : depending on the number of lines, accordingly I have written the code in the next function to determine whether its single or paired, followed by additional check ( if there are two files provided by the user ), to match every ReadID in both the files.

Reply:
For input files, I would still prefer to stick to FASTQ format _only_ for now. So SRA doesn't come into the picture, just different formats of ID lines _within a FASTQ file_.

Reply:
Whether it will make sense in the future to support SRA files as input will depend on whether we could run the whole package on SRA files or not. It's not totally out of the question, but it's not sth that has a high priority right now and we would prefer to focus on the basic implementation for now before adding features that we don't really need.